### PR TITLE
Liberty dev mode integration

### PR DESCRIPTION
**Pull request by awisniew90**
1. Allow users to set boost properties in maven pom
2. Collect boost properties once at the beginning of mojo execution
3. Have Liberty generate config variables rather than bootstrap.properties
4. Change Liberty server name from BoostServer to defaultServer
5. Move password encryption to LibertyServerConfigGenerator (since Liberty specific)

### CODE REVIEW

File: boost-maven/boost-maven-plugin/src/main/java/boost/maven/plugin/AbstractMojo.java [hunk 1]
```diff
@@ -103,37 +105,41 @@ public void execute() throws MojoExecutionException {
...
 
     protected RuntimeI getRuntimeInstance() throws MojoExecutionException {
-        if (runtime == null) {
-            RuntimeParams params = new RuntimeParams(boosterConfigs, getExecutionEnvironment(), project, getLog(),
-                    repoSystem, repoSession, remoteRepos, getMavenDependencyPlugin());
-            try {
-                ServiceLoader<RuntimeI> runtimes = ServiceLoader.load(RuntimeI.class, projectClassLoader);
-                if (!runtimes.iterator().hasNext()) {
+        
+    	RuntimeI runtime = null;
```
[the remark this answers is not in the document]
Comment (awisniew90) [reply boost-maven/boost-maven-plugin/src/main/java/boost/maven/plugin/AbstractMojo.java]: Fixed

File: boost-common/src/main/java/boost/common/boosters/CDIBoosterConfig.java [hunk 2]
```diff
@@ -21,7 +22,7 @@
 @BoosterCoordinates(AbstractBoosterConfig.BOOSTERS_GROUP_ID + ":cdi")
 public class CDIBoosterConfig extends AbstractBoosterConfig {
 
-    public CDIBoosterConfig(Map<String, String> dependencies, BoostLoggerI logger) throws BoostException {
+    public CDIBoosterConfig(Map<String, String> dependencies, Properties boostProperties, BoostLoggerI logger) throws BoostException {
```
[the remark this answers is not in the document]
Comment (awisniew90) [reply boost-common/src/main/java/boost/common/boosters/CDIBoosterConfig.java]: Added this in

**Comment by awisniew90** [comment]
@scottkurz - I created a BoosterConfigParams class to use when passing parameters to BoosterConfig classes (rather than the separate dependencies and properties).  Fixed DB2 tests as well.

File: boost-maven/boost-maven-plugin/src/it/test-jdbc-db2/invoker.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
 #Using profiles to configure failsafe to run certain tests
-invoker.goals.1 = clean install -PnoEncoding -Dboost.db.databaseName=myCustomDB -Dboost.db.user=user -Dboost.db.password=password -Dboost.aes.key=test
+invoker.goals.1 = clean install
```
Comment (scottkurz) [inline boost-maven/boost-maven-plugin/src/it/test-jdbc-db2/invoker.properties]: Did we lose some coverage here?

**Comment by scottkurz** [comment]
Resolved by #361 so closing this as dup.

**Pull request closed without merging**